node/src/Pos/Delegation/Logic/VAR.hs

```diff
@@ -556,23 +560,33 @@ dlgRollbackBlocks blunds = do
         let postedOp = SomeBatchOp $ map (GS.DelPostedThisEpoch . addressHash) issuers
         pure $ pskOp <> postedOp
 
--- | Normalizes the memory state after the rollback.
+-- | Normalizes the memory state after the rollback. Must be called
+-- only when 'StateLock' is taken.
 dlgNormalizeOnRollback ::
        forall ssc ctx m.
        ( MonadDelegation ctx m
        , DB.MonadBlockDB ssc m
        , DB.MonadGState m
        , MonadIO m
        , MonadMask m
-       , HasLens LrcContext ctx LrcContext
+       , HasLens' ctx LrcContext
        , Mockable CurrentTime m
+       , WithLogger m
+       , HasCoreConstants
+       , SscHelpersClass ssc
        )
     => m ()
 dlgNormalizeOnRollback = do
     tip <- DB.getTipHeader @ssc
     oldPool <- runDelegationStateAction $ do
-        dwEpochId .= (tip ^. epochIndexL)
         pool <- uses dwProxySKPool toList
         dwProxySKPool .= mempty
         pure pool
-    forM_ oldPool (processProxySKHeavy @ssc)
+    forM_ oldPool $ \psk -> do
+        res <- processProxySKHeavyInternal @ssc psk
+        -- should we throw error here also?
```

Comment:
No, we shouldn't throw error. Why do we even log it as error? If we have a PSK in mempool and it becomes invalid after rollback, it's absolutely legal situation, we can ignore it. That's what normalization does.